core/templates/pages/contributor-dashboard-page/translation-language-selector/translation-language-selector.component.ts

```diff
@@ -89,17 +89,32 @@ export class TranslationLanguageSelectorComponent implements OnInit {
       this.languageIdToDescription[this.activeLanguageCode] :
       'Select a language...'
     );
+
+    this.contributionOpportunitiesBackendApiService
```

Comment:
I'm a bit confused about why you're doing this in a separate call to the backend.

Couldn't it be combined with the initial call that's used to fetch the data for the CD? Otherwise it increases the overall loading/stabilization time for the page.

Reply:
I am not sure how we can do that? I mean contributor dashboard does not directly load to the "Translate Text" tab so we do not call for any data related to the "Translate Text" tab initially and only load data for "My Contributions" tab. When the user switches the tab then only we render this translate-language-selector component and call for the preferred translation language.
Should I make this call from somewhere else? If yes, then from where and how I will be able to get the data over here in this component ( translate-language-selector ) because at the end I need to change the value of this component so that it should further call for the opportunities available in that language.

Reply:
Hm, I'm looking again at the overall code for the contributor dashboard, and I actually see this pattern quite a bit -- lots of small calls for little pieces of data. So I suppose this follows the existing trend and is probably OK, given the context.

@sagangwee -- just wondering though, have you considered this point and do you think this might be impacting page speed?

Reply:
That's an interesting point. I suppose we could batch together any CD related data on initial page component load into one API endpoint, e.g. combine this fetch with the `ContributionOpportunitiesService.getTranslatableTopicNamesAsync()` [here](https://github.com/oppia/oppia/blob/develop/core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts#L216) and then fanout the data to the child components.

Though this small optimization might be dwarfed by all the other network requests we're making, e.g. we make at least 3 calls to `UserService` in the [contributor dashboard page component](https://github.com/oppia/oppia/blob/develop/core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts#L162).

It also looks like we're repeating a bunch of network requests on initial page load, e.g. multiple calls to `/gettranslatabletopicnames`:
![Screen Shot 2022-06-14 at 6 05 22 PM](https://user-images.githubusercontent.com/9004726/173698031-7f67afb3-9ede-45d5-b625-c2a545093892.png)

So there might be some Angular improvements we can make there.

Assessing the performance impact probably needs a bit of work, so I'm inclined to leave the API usage as is until we can get better impact measurements.